### PR TITLE
Implement tod_manager::resolve_random() without Boost.Range

### PR DESCRIPTION
**Pull request by jyrkive**
This gets rid of yet another use of Boost.Bind. After this, only server.cpp remains. :smiling_imp:

### CODE REVIEW

**Comment by jyrkive** [comment]
Okay, I got it fixed. The remaining errored build is a random test failure.

**Comment by gfgtdf** [comment]
hmm what exactly are the advnatage of these changes?

**Comment by Vultraz** [comment]
@gfgtdf removing the use of boost::bind

**Comment by gfgtdf** [comment]
if you want thatt couldn't you just palce it with std::bind or a lambda then?

**Comment by jyrkive** [comment]
@gfgtdf Merely replacing it with `std::bind()` doesn't compile. I guess Boost.Range is incompatible with `std::bind()`.

https://gist.github.com/jyrkive/fcc4c336b3e85e27d73b61d683a937c2

**Comment by gfgtdf** [comment]
Still i wonder why you want to replace boost::bind in the first place?

**Comment by CelticMinstrel** [comment]
Oh uhh... I actually did the same thing without realizing you had done it. Functionally I think my version is pretty much the same, though it uses std::bind instead of lambdas and sticks to lexical_cast_default instead of switching to std::stoi. The latter is something of a sticking point; I'm not quite sure which way is better, and they're significantly different in semantics.

**Comment by Vultraz** [comment]
lexical_cast_default just means there's a fallback default value. Also, I dunno if bind is inherently better than a lambda.

**Comment by CelticMinstrel** [comment]
> lexical_cast_default just means there's a fallback default value.

More importantly than that, it means no exception will be thrown.

> Also, I dunno if bind is inherently better than a lambda.

I don't think either of the two are better than the other.

BTW, for reference, this is what my implementation looks like:

```C++
std::vector<std::string> input = utils::split(random_tod_.str());
std::vector<int> output;
std::transform(input.begin(), input.end(), std::back_inserter(output), std::bind(lexical_cast_default<int, std::string>, _1, 0));
auto new_end = std::remove_if(output.begin(), output.end(), std::bind(std::less_equal<int>(), _1, 0));
output.erase(new_end, output.end());
```

**Comment by jyrkive** [comment]
I intentionally switched to `std::stoi()`. It's the standard C++11 way to convert a string to an integer.

As for why I used lambdas, in both cases the game didn't compile when I tried to use `std::bind()`. With `std::stoi()` the compiler complained that it doesn't know which overload to select.

With comparison to zero, it's my fault. (Thanks @CelticMinstrel  for showing me the correct syntax.) I had written

    std::bind(std::less_equal<int>, std::placeholders::_1, 0)

instead of

    std::bind(std::less_equal<int>(), std::placeholders::_1, 0)

In other words, I accidentally attempted to pass the type, not an instance.

I think I won't switch it to `std::bind()` at this point because the difference between the options is so small that it doesn't really matter.

**Comment by CelticMinstrel** [comment]
> It's the standard C++11 way to convert a string to an integer.

The problem here is that it throws an exception, which needs to be accounted for somehow.

**Comment by jyrkive** [comment]
> The problem here is that it throws an exception, which needs to be accounted for somehow.

I do account for it (second commit), so I don't really see a problem.

**Comment by CelticMinstrel** [comment]
(For the record - I initially ignored the second commit, which is why I missed the exception handling.)
